### ISSUE TITLE
startup analysis process took 5 ~ 10 seconds

### ISSUE THREAD

**Issue by GopherJ** [rust-analyzer/rust-analyzer#5109]
Hi, I'm using rust-analyzer through coc.nvim and coc-rust-analyzer. I found that when I open a new rust project, it took too much time to finish analyzing.

I posted then a question for coc-rust-analyzer: https://github.com/fannheyward/coc-rust-analyzer/issues/299

The author told me it's more a problem on `rust-analyzer` side.

So summary of the problem is:

> can the startup analysis process be optimized?  now it takes 5 ~ 10 seconds, that means when I open a new rust project I'll need to wait for 5 to 10 seconds before starting to work.

Also the project I used to test isn't really too big.. around 10+ files around 150+ lines per file

**Comment by GopherJ** [comment]
Also there isn't any cache, when I open the project and then exit and enter again. it needs to do the initial analysis again....

**Comment by bjorn3** [comment]
Do you have any dependencies? If so all of them will need to be analyzed before the current project will be analyzed. In any case libcore, liballoc and libstd need to be analyzed.

re caching: this is #4712

**Comment by GopherJ** [comment]
```

[package]
name = "sda_server"
version = "2.0.0"
authors = ["Cheng JIANG <jiang.cheng@vip.163.com>"]
edition = "2018"
description = "Social distancing api server"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
actix-web = { version = "2.0.0", default-features = false, features = ["rustls", "compress"] }
env_logger = "0.7.1"
log = "0.4.8"
thiserror = "1.0.15"
actix-web-actors = "2.0.0"
actix-rt = "1.1.0"
clap = { version = "2.33.0", features = ["suggestions", "color"] }
lazy_static = "1.4.0"
actix = "0.9.0"
serde = { version = "1.0.90", features = ["derive", "rc"] }
serde_json = "1.0.39"
futures = { version = "0.3.4", features = ["io-compat"] }
backoff = "0.1.6"
rust-embed = "5.5.1"
mime_guess = "2.0.3"
tokio = { version = "0.2.13", features = ["net", "fs", "sync", "io-util", "rt-core", "blocking", "rt-threaded", "time"] }
actix-cors = "0.2.0"
uuid = { version = "0.8.1", features = ["v4"] }
config = "0.9"
elasticsearch = { version = "7.7.0-alpha.1", default-features = false, features = ["rustls-tls"] }
url = "2.1.1"
redis = "0.15"
reqwest = { version = "0.10.4", features = ["json", "stream", "rustls-tls", "gzip"] }
ttl_cache = "0.5.1"
toml = "0.5.6"
diesel = { version = "1.4.2", features = ["r2d2", "postgres", "chrono", "uuid"] }
actix-web-httpauth = { version = "0.4.0", path = "third-party/actix-web-httpauth" }
r2d2 = "0.8.8"
derive_builder = "0.9.0"
chrono = { version = "0.4.10", features = ["serde"] }
# make diesel being able to be compiled through musl
openssl = "0.10.29"
openssl-probe = "0.1.2"

```

**Comment by bjorn3** [comment]
With that many dependencies 5-10 seconds of analysis time is quite good in my opinion. This leaves adding a cache:

Duplicate of #4712
